diff -rupN fs_original/bin/mountSD fs/bin/mountSD
--- fs_original/bin/mountSD	2014-05-21 00:28:38.725664742 +0200
+++ fs/bin/mountSD	2014-05-21 00:33:22.260252879 +0200
@@ -1,10 +1,10 @@
 #!/bin/sh
 echo "Starte mountSD: $@" > /tmp/dudu
-mount $@
-if [ $? == 0 ] ; then
+# mount $@
+# if [ $? == 0 ] ; then
 	touch /var/status/SDmounted
 	if [ -f /media/sd-mmcblk0/.initialised ] ; then
 		touch /var/status/SDinitialised
 	fi
-fi
+# fi
 
diff -rupN fs_original/bin/update_firmware_pre fs/bin/update_firmware_pre
--- fs_original/bin/update_firmware_pre	2014-05-21 00:28:38.845678386 +0200
+++ fs/bin/update_firmware_pre	2014-05-21 00:33:22.270253669 +0200
@@ -1,5 +1,5 @@
 #!/bin/sh
-
+##### TODO LXCCU: include rcu installer here or something like that
 log() {
   logger -t homematic -p user.notice $1
 }
@@ -16,47 +16,47 @@ if [ ! -f /var/new_firmware.tar.gz ]; th
 	exit
 fi
 
-echo Extract firmware update file ...
-
-# Info LED
-echo timer > /sys/class/leds/info/trigger
-echo 255 > /sys/class/leds/info/brightness
-echo 100 > /sys/class/leds/info/delay_off
-echo 100 > /sys/class/leds/info/delay_on
-
-# Power LED
-echo timer > /sys/class/leds/power/trigger
-echo 255 > /sys/class/leds/power/brightness
-echo 100 > /sys/class/leds/power/delay_off
-echo 100 > /sys/class/leds/power/delay_on
-
-cd /var
-
-cat new_firmware.tar.gz | gunzip | tar x
-
-if [ ! -x /var/update_script ]; then
-        log "Error unzipping firmware image archive. Rebooting"
-        echo "Error unzipping firmware image archive. Rebooting"
-        # echo x > /dev/watchdog
-        #/sbin/reboot
-        #while true ; do true ;  done
-        exit
-fi
-
-rm new_firmware.tar.gz
-
-echo Copy files...
+# echo Extract firmware update file ...
 
-cp -a /bin /var
-cp -a /lib /var
-cp -a /sbin /var
-cp -a /etc /var
-cp -a /usr /var
-cp -a /dev /var
-mkdir /var/proc
-mkdir /var/sys
+# # Info LED
+# echo timer > /sys/class/leds/info/trigger
+# echo 255 > /sys/class/leds/info/brightness
+# echo 100 > /sys/class/leds/info/delay_off
+# echo 100 > /sys/class/leds/info/delay_on
+
+# # Power LED
+# echo timer > /sys/class/leds/power/trigger
+# echo 255 > /sys/class/leds/power/brightness
+# echo 100 > /sys/class/leds/power/delay_off
+# echo 100 > /sys/class/leds/power/delay_on
+
+# cd /var
+
+# cat new_firmware.tar.gz | gunzip | tar x
+
+# if [ ! -x /var/update_script ]; then
+#         log "Error unzipping firmware image archive. Rebooting"
+#         echo "Error unzipping firmware image archive. Rebooting"
+#         # echo x > /dev/watchdog
+#         #/sbin/reboot
+#         #while true ; do true ;  done
+#         exit
+# fi
+
+# rm new_firmware.tar.gz
+
+# echo Copy files...
+
+# cp -a /bin /var
+# cp -a /lib /var
+# cp -a /sbin /var
+# cp -a /etc /var
+# cp -a /usr /var
+# cp -a /dev /var
+# mkdir /var/proc
+# mkdir /var/sys
 
-mkdir oldroot
+# mkdir oldroot
 
-pivot_root . oldroot
+# pivot_root . oldroot
 
diff -rupN fs_original/bin/update_firmware_run fs/bin/update_firmware_run
--- fs_original/bin/update_firmware_run	2014-05-21 00:28:38.845678386 +0200
+++ fs/bin/update_firmware_run	2014-05-21 00:33:22.270253669 +0200
@@ -1,36 +1,70 @@
 #!/bin/sh
 
-mount /proc
-mount /sys
+# mount /proc
+# mount /sys
 
-if [ ! -f /update_script ]; then
-  echo "check Factory reset"
-  if [ -f /var/doFactoryReset ]; then
-	/etc/init.d/S15watchdog start
-	echo "do Factory reset"
-	umount /usr/local
-        /usr/sbin/ubidetach -p /dev/mtd6
-        /usr/sbin/ubiformat /dev/mtd6 -y
-        /usr/sbin/ubiattach -p /dev/mtd6
-        /usr/sbin/ubimkvol /dev/ubi1 -N user -m
-	sleep 5
-  fi
-  echo 1 > /proc/sys/kernel/sysrq
-  echo b > /proc/sysrq-trigger
+# if [ ! -f /update_script ]; then
+#   echo "check Factory reset"
+#   if [ -f /var/doFactoryReset ]; then
+#   /etc/init.d/S15watchdog start
+#   echo "do Factory reset"
+#   umount /usr/local
+#         /usr/sbin/ubidetach -p /dev/mtd6
+#         /usr/sbin/ubiformat /dev/mtd6 -y
+#         /usr/sbin/ubiattach -p /dev/mtd6
+#         /usr/sbin/ubimkvol /dev/ubi1 -N user -m
+#   sleep 5
+#   fi
+#   echo 1 > /proc/sys/kernel/sysrq
+#   echo b > /proc/sysrq-trigger
+# fi
+
+# umount /oldroot/dev/pts
+# umount /oldroot/dev/shm
+# umount /oldroot/dev
+# umount /oldroot/tmp/
+# umount /oldroot/sys/kernel/debug
+# umount /oldroot/sys
+# umount /oldroot/proc
+# umount /oldroot/usr/local
+# umount /oldroot/media/sd-mmcblk0
+# umount /oldroot/media
+# umount /oldroot
+# /etc/init.d/S15watchdog start
+# /update_script CCU2
+# echo 1 > /proc/sys/kernel/sysrq 
+# echo b > /proc/sysrq-trigger
+
+log() {
+  logger -t homematic -p user.notice $1
+}
+
+echo "LXCCU: Entering addon install mode"
+log "LXCCU: Entering addon install mode"
+if [ ! -f /var/new_firmware.tar.gz ]; then
+  log "LXCCU: addon image archive does not exist. Nothing to do"
+  echo "LXCCU: addon image archive does not exist. Nothing to do"
+  exit
 fi
 
-umount /oldroot/dev/pts
-umount /oldroot/dev/shm
-umount /oldroot/dev
-umount /oldroot/tmp/
-umount /oldroot/sys/kernel/debug
-umount /oldroot/sys
-umount /oldroot/proc
-umount /oldroot/usr/local
-umount /oldroot/media/sd-mmcblk0
-umount /oldroot/media
-umount /oldroot
-/etc/init.d/S15watchdog start
-/update_script CCU2
-echo 1 > /proc/sys/kernel/sysrq 
-echo b > /proc/sysrq-trigger
+echo "LXCCU: extract addon archive"
+log "LXCCU: extract addon archive"
+cd /var
+cat new_firmware.tar.gz | gunzip | tar x
+rm new_firmware.tar.gz
+if [ ! -x /var/update_script ]; then
+  log "LXCCU: Error unzipping addon image archive. Nothing to do"
+  echo "LXCCU: Error unzipping addon image archive. Nothing to do"
+  exit
+fi
+
+log "LXCCU: prepare mount points to simulate ccu"
+echo "LXCCU: prepare mount points to simulate ccu"
+cat /var/update_script | sed -r 's/mount -t ubifs ubi0:root ([\/\w\-]*)/mount -o bind \/ \1/' > /var/update_script_prepare
+cat /var/update_script_prepare | sed -r 's/mount -t ubifs ubi1:user ([\/\w\-]*)/mount -o bind \/usr\/local \1/' > /var/update_script
+
+log "LXCCU: start update_script as LXCCU"
+echo "LXCCU: start update_script as LXCCU"
+chmod +x /var/update_script
+cd /var/
+/var/update_script CCU2
diff -rupN fs_original/etc/config_templates/ntpclient fs/etc/config_templates/ntpclient
--- fs_original/etc/config_templates/ntpclient	2014-05-21 00:28:40.395854612 +0200
+++ fs/etc/config_templates/ntpclient	2014-05-21 00:33:22.280254459 +0200
@@ -1 +1 @@
-NTPSERVERS='ntp.homematic.com'
+NTPSERVERS='0.de.pool.ntp.org'
diff -rupN fs_original/etc/fstab fs/etc/fstab
--- fs_original/etc/fstab	2014-05-21 00:28:40.415856886 +0200
+++ fs/etc/fstab	2014-05-21 00:33:22.280254459 +0200
@@ -1,13 +1,3 @@
 # /etc/fstab: static file system information.
 #
 # <file system> <mount pt>     <type>   <options>         <dump> <pass>
-/dev/root       /              ext2     rw,noauto         0      1
-proc            /proc          proc     defaults          0      0
-devpts          /dev/pts       devpts   defaults,gid=5,mode=620   0      0
-tmpfs           /dev/shm       tmpfs    mode=0777         0      0
-tmpfs           /tmp           tmpfs    defaults          0      0
-mediafs         /media         tmpfs    defaults          0      0
-sysfs           /sys           sysfs    defaults          0      0
-varfs           /var           tmpfs    defaults,size=196M 0      0
-debugfs 	/sys/kernel/debug debugfs 		  0 	0
-ubi1:user	/usr/local	ubifs	defaults,noatime,rw,sync	0 0
diff -rupN fs_original/etc/init.d/_40UsbNetwork fs/etc/init.d/_40UsbNetwork
--- fs_original/etc/init.d/_40UsbNetwork	1970-01-01 01:00:00.000000000 +0100
+++ fs/etc/init.d/_40UsbNetwork	2014-05-21 00:33:22.290255249 +0200
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Starts Ethernet over USB
+#
+
+
+start() {
+	export TZ=`cat /etc/config/TZ`
+
+	echo -n "Starting network over usb: "
+		
+	if [ ! -f /etc/udhcpd.usb0.conf ]; then
+		echo "ERROR (file /etc/udhcp.usb0.conf not found)"
+	elif [ ! -f /etc/dnsd.conf ]; then
+		echo "ERROR (file /etc/dnsd.conf not found)"
+	else
+		ifconfig usb0 10.101.82.51 netmask 255.255.255.0
+		udhcpd -S /etc/udhcpd.usb0.conf
+		# disable DNS Server; CCU can only be access by ip address
+		#dnsd -d -c /etc/dnsd.conf -i 10.101.82.51
+		echo "OK"	
+	fi		
+}
+
+stop() {
+	echo -n "Stopping network over usb: "
+	killall udhcpd
+	killall dnsd	
+	echo "OK"
+}
+restart() {
+	stop
+	start
+}
+
+case "$1" in
+  start)
+	start
+	;;
+  stop)
+	stop
+	;;
+  restart|reload)
+	restart
+	;;
+  *)
+	echo "Usage: $0 {start|stop|restart}"
+	exit 1
+esac
+
+exit $?
+
diff -rupN fs_original/etc/init.d/rcS fs/etc/init.d/rcS
--- fs_original/etc/init.d/rcS	2014-05-21 00:28:40.265839833 +0200
+++ fs/etc/init.d/rcS	2014-05-21 00:33:22.300256039 +0200
@@ -25,3 +25,4 @@ for i in /etc/init.d/S??* ;do
     esac
 done
 
+/bin/mountSD
\ No newline at end of file
diff -rupN fs_original/etc/init.d/S00eQ3SystemStart fs/etc/init.d/S00eQ3SystemStart
--- fs_original/etc/init.d/S00eQ3SystemStart	2014-05-21 00:28:40.305844380 +0200
+++ fs/etc/init.d/S00eQ3SystemStart	2014-05-21 00:33:22.300256039 +0200
@@ -4,104 +4,104 @@
 #
 
 CFG_TEMPLATE_DIR=/etc/config_templates
-SERIAL=$(cat /sys/module/plat_eq3ccu2/parameters/board_serial)
+#SERIAL=$(cat /sys/module/plat_eq3ccu2/parameters/board_serial)
 
 init() {
 	# power led on
-	echo 255 > /sys/class/leds/power/brightness
-	echo default-on > /sys/class/leds/power/trigger
+#	echo 255 > /sys/class/leds/power/brightness
+#	echo default-on > /sys/class/leds/power/trigger
 	# internet led off
-	echo none > /sys/class/leds/internet/trigger
+#	echo none > /sys/class/leds/internet/trigger
 	# info led fast
-	echo timer > /sys/class/leds/info/trigger
-	echo 255 > /sys/class/leds/info/brightness
-	echo 100 > /sys/class/leds/info/delay_off
-	echo 100 > /sys/class/leds/info/delay_on
-
-        chmod 775 /var
-        mkdir /var/log
-        chmod 775 /var/log
-        mkdir /var/tmp
-        chmod 775 /var/tmp
-        mkdir /var/rega
-        chmod 775 /var/rega
-        mkdir /var/run
-        chmod 775 /var/run
-        mkdir /var/spool
-        chmod 775 /var/spool
-        mkdir /var/lock
-        chmod 775 /var/lock
-        mkdir /var/cache
-        chmod 775 /var/cache
-        mkdir /var/lib
-        chmod 775 /var/lib
-        mkdir /var/lib/misc
-        chmod 775 /var/lib/misc
-        mkdir /var/empty
-        chmod 600 /var/empty
-        mkdir /var/etc
-        chmod 775 /var/etc
+#	echo timer > /sys/class/leds/info/trigger
+#	echo 255 > /sys/class/leds/info/brightness
+#	echo 100 > /sys/class/leds/info/delay_off
+#	echo 100 > /sys/class/leds/info/delay_on
+
+	       chmod 775 /var
+	       mkdir /var/log
+	       chmod 775 /var/log
+	       mkdir /var/tmp
+	       chmod 775 /var/tmp
+	       mkdir /var/rega
+	       chmod 775 /var/rega
+	       mkdir /var/run
+	       chmod 775 /var/run
+	       mkdir /var/spool
+	       chmod 775 /var/spool
+	       mkdir /var/lock
+	       chmod 775 /var/lock
+	       mkdir /var/cache
+	       chmod 775 /var/cache
+	       mkdir /var/lib
+	       chmod 775 /var/lib
+	       mkdir /var/lib/misc
+	       chmod 775 /var/lib/misc
+	       mkdir /var/empty
+	       chmod 600 /var/empty
+	       mkdir /var/etc
+	       chmod 775 /var/etc
 	#
 	# crontab dirs
-        mkdir -p /var/spool/cron
-        chmod 775 /var/spool
-        chmod 775 /var/spool/cron
-        if [ ! -d /usr/local/crontabs ] ; then
-        	mkdir -p /usr/local/crontabs
-        	chmod 775 /usr/local/crontabs
-        fi
+	       mkdir -p /var/spool/cron
+	       chmod 775 /var/spool
+	       chmod 775 /var/spool/cron
+	       if [ ! -d /usr/local/crontabs ] ; then
+	       	mkdir -p /usr/local/crontabs
+	       	chmod 775 /usr/local/crontabs
+	       fi
 	ln -s /usr/local/crontabs /var/spool/cron/crontabs 
-        if [ ! -e /usr/local/crontabs/root ] ; then
-                cp $CFG_TEMPLATE_DIR/crontab.root /usr/local/crontabs/root
-        fi
-        # Bug fix
-        grep "setInterfaceClock" /usr/local/crontabs/root
-        if [ $? == 0 ] ; then
-                echo "update crontab file"
-                sed 's/setInterfaceClock/SetInterfaceClock/g' /usr/local/crontabs/root > /usr/local/crontabs/root.tmp
-                mv /usr/local/crontabs/root.tmp /usr/local/crontabs/root
-        fi
+	       if [ ! -e /usr/local/crontabs/root ] ; then
+	               cp $CFG_TEMPLATE_DIR/crontab.root /usr/local/crontabs/root
+	       fi
+	       # Bug fix
+	       grep "setInterfaceClock" /usr/local/crontabs/root
+	       if [ $? == 0 ] ; then
+	               echo "update crontab file"
+	               sed 's/setInterfaceClock/SetInterfaceClock/g' /usr/local/crontabs/root > /usr/local/crontabs/root.tmp
+	               mv /usr/local/crontabs/root.tmp /usr/local/crontabs/root
+	       fi
 
-        
+	       
 	if [ ! -e /usr/local/etc/config/shadow ] ; then
 		cp $CFG_TEMPLATE_DIR/shadow /usr/local/etc/config/shadow
 	fi
 
-	modprobe spidev
-        modprobe ic200_spi
-        modprobe spi_eq3_gpio
-        modprobe spi_bitbang
-        mknod spidev0.0 c 153 0
-        modprobe gpio-keys
-        modprobe fsl_usb2_udc.ko
-	if grep -q "eQ3Mode=production" /proc/cmdline ; then
-          modprobe g_multi luns=1 cdrom=1 ro=1 removable=0,0 file=/usr/share/eq3/install.iso host_addr=00:1a:22:00:05:86 dev_addr=00:1a:22:00:05:87 iManufacturer="eQ-3" iProduct="CCU2" idVendor="0x1b1f" idProduct="0xc016" iSerialNumber="$SERIAL"
-	fi
+#	modprobe spidev
+#	       modprobe ic200_spi
+#	       modprobe spi_eq3_gpio
+#	       modprobe spi_bitbang
+#	       mknod spidev0.0 c 153 0
+#	       modprobe gpio-keys
+#	       modprobe fsl_usb2_udc.ko
+#	if grep -q "eQ3Mode=production" /proc/cmdline ; then
+#	         modprobe g_multi luns=1 cdrom=1 ro=1 removable=0,0 file=/usr/share/eq3/install.iso host_addr=00:1a:22:00:05:86 dev_addr=00:1a:22:00:05:87 iManufacturer="eQ-3" iProduct="CCU2" idVendor="0x1b1f" idProduct="0xc016" iSerialNumber="$SERIAL"
+#	fi
 	
 	# Tunneling
-        modprobe tun
-       
+#	       modprobe tun
+	      
 	# USB
-        modprobe ehci-hcd
+#	       modprobe ehci-hcd
 	 
 	if [ ! -d /usr/local/etc/config ] ; then
-                mkdir -p /usr/local/etc/config
-        	chmod 775 /usr/local/etc/config
-        fi
+	               mkdir -p /usr/local/etc/config
+	       	chmod 775 /usr/local/etc/config
+	       fi
 
 
-        if [ ! -e /etc/config/TZ ] ; then
-                cp $CFG_TEMPLATE_DIR/TZ /etc/config
-        fi
-        if [ ! -e /etc/config/netconfig ] ; then
-                cp $CFG_TEMPLATE_DIR/netconfig /etc/config
-        fi
-        if [ ! -e /etc/config/shadow ] ; then
-                cp $CFG_TEMPLATE_DIR/shadow /etc/config
-        fi
+	       if [ ! -e /etc/config/TZ ] ; then
+	               cp $CFG_TEMPLATE_DIR/TZ /etc/config
+	       fi
+	       if [ ! -e /etc/config/netconfig ] ; then
+	               cp $CFG_TEMPLATE_DIR/netconfig /etc/config
+	       fi
+	       if [ ! -e /etc/config/shadow ] ; then
+	               cp $CFG_TEMPLATE_DIR/shadow /etc/config
+	       fi
 
 	export TZ=`cat /etc/config/TZ`
- 
+
 }
 
 start() {
@@ -116,6 +116,7 @@ start() {
 stop () {
 	start-stop-daemon -K -q -p /var/run/crond.pid
 	/bin/update_firmware_pre
+	/bin/update_firmware_run
 }
 
 restart() {
@@ -123,16 +124,16 @@ restart() {
 }
 
 case "$1" in
-  start)
+	 start)
 	start
 	;;
-  stop)
+	 stop)
 	stop
 	;;
-  restart|reload)
+	 restart|reload)
 	restart
 	;;
-  *)
+	 *)
 	echo "Usage: $0 {start|stop|restart}"
 	exit 1
 esac
diff -rupN fs_original/etc/init.d/S40UsbNetwork fs/etc/init.d/S40UsbNetwork
--- fs_original/etc/init.d/S40UsbNetwork	2014-05-21 00:28:40.305844380 +0200
+++ fs/etc/init.d/S40UsbNetwork	1970-01-01 01:00:00.000000000 +0100
@@ -1,52 +0,0 @@
-#!/bin/sh
-#
-# Starts Ethernet over USB
-#
-
-
-start() {
-	export TZ=`cat /etc/config/TZ`
-
-	echo -n "Starting network over usb: "
-		
-	if [ ! -f /etc/udhcpd.usb0.conf ]; then
-		echo "ERROR (file /etc/udhcp.usb0.conf not found)"
-	elif [ ! -f /etc/dnsd.conf ]; then
-		echo "ERROR (file /etc/dnsd.conf not found)"
-	else
-		ifconfig usb0 10.101.82.51 netmask 255.255.255.0
-		udhcpd -S /etc/udhcpd.usb0.conf
-		# disable DNS Server; CCU can only be access by ip address
-		#dnsd -d -c /etc/dnsd.conf -i 10.101.82.51
-		echo "OK"	
-	fi		
-}
-
-stop() {
-	echo -n "Stopping network over usb: "
-	killall udhcpd
-	killall dnsd	
-	echo "OK"
-}
-restart() {
-	stop
-	start
-}
-
-case "$1" in
-  start)
-	start
-	;;
-  stop)
-	stop
-	;;
-  restart|reload)
-	restart
-	;;
-  *)
-	echo "Usage: $0 {start|stop|restart}"
-	exit 1
-esac
-
-exit $?
-
diff -rupN fs_original/etc/init.d/S50eq3configd fs/etc/init.d/S50eq3configd
--- fs_original/etc/init.d/S50eq3configd	2014-05-21 00:28:40.265839833 +0200
+++ fs/etc/init.d/S50eq3configd	2014-05-21 00:33:22.320257619 +0200
@@ -1,52 +1,52 @@
 #!/bin/sh
 #
-# Starts eq3configd.
+#	Starts	eq3configd.
 #
-init () {
-	radio_mac=$(cat /sys/module/plat_eq3ccu2/parameters/radio_mac)
-	board_serial=$(cat /sys/module/plat_eq3ccu2/parameters/board_serial)
-	echo "BidCoS-Address=$radio_mac" > /var/ids
-	echo "SerialNumber=$board_serial" >> /var/ids
-	if [ ! -e /etc/config/ids ] ; then
-    		cp /var/ids /etc/config
+init	()	{
+#	radio_mac=$(cat	/sys/module/plat_eq3ccu2/parameters/radio_mac)
+#	board_serial=$(cat	/sys/module/plat_eq3ccu2/parameters/board_serial)
+	echo	"BidCoS-Address=$radio_mac"	>	/var/ids
+	echo	"SerialNumber=$board_serial"	>>	/var/ids
+	if	[	!	-e	/etc/config/ids	]	;	then
+						cp	/var/ids	/etc/config
 	fi
 	
-	if [ ! -e /etc/config/crypttool.cfg ] ; then
-			touch /etc/config/crypttool.cfg
+	if	[	!	-e	/etc/config/crypttool.cfg	]	;	then
+			touch	/etc/config/crypttool.cfg
 	fi
 }
 
-start() {
-	echo -n "Starting eq3configd: "
+start()	{
+	echo	-n	"Starting	eq3configd:	"
 	init
-	start-stop-daemon -S -q -p /var/run/eq3configd.pid --exec /bin/eq3configd &
-	start-stop-daemon -S -q -p /var/run/ssdpd.pid --exec /bin/ssdpd &
-	echo "OK"
+	start-stop-daemon	-S	-q	-p	/var/run/eq3configd.pid	--exec	/bin/eq3configd	&
+	start-stop-daemon	-S	-q	-p	/var/run/ssdpd.pid	--exec	/bin/ssdpd	&
+	echo	"OK"
 }
-stop() {
-	echo -n "Stopping eq3configd: "
-	start-stop-daemon -K -q -p /var/run/eq3configd.pid
-	echo "OK"
+stop()	{
+	echo	-n	"Stopping	eq3configd:	"
+	start-stop-daemon	-K	-q	-p	/var/run/eq3configd.pid
+	echo	"OK"
 }
-restart() {
+restart()	{
 	stop
 	start
 }
 
-case "$1" in
-  start)
+case	"$1"	in
+		start)
 	start
 	;;
-  stop)
+		stop)
 	stop
 	;;
-  restart|reload)
+		restart|reload)
 	restart
 	;;
-  *)
-	echo "Usage: $0 {start|stop|restart}"
-	exit 1
+		*)
+	echo	"Usage:	$0	{start|stop|restart}"
+	exit	1
 esac
 
-exit $?
+exit	$?
 
diff -rupN fs_original/etc/init.d/S60rfd fs/etc/init.d/S60rfd
--- fs_original/etc/init.d/S60rfd	2014-05-21 00:28:40.325846654 +0200
+++ fs/etc/init.d/S60rfd	2014-05-21 00:33:22.320257619 +0200
@@ -25,31 +25,31 @@ init() {
 	export TZ=`cat /etc/config/TZ`
 
 	if [ ! -d /etc/config/rfd ] ; then
-  		mkdir /etc/config/rfd
+	 		mkdir /etc/config/rfd
 	fi
 	if [ ! -e /etc/config/rfd.conf ] ; then
 		cp $CFG_TEMPLATE_DIR/rfd.conf /etc/config
 	fi
 
-        # Bug fix
-        grep "/etc/config/rfd/keys" /etc/config/rfd.conf
-        if [ $? == 0 ] ; then
-                echo "Copy rfd files"
+	       # Bug fix
+	       grep "/etc/config/rfd/keys" /etc/config/rfd.conf
+	       if [ $? == 0 ] ; then
+	               echo "Copy rfd files"
 		cp $CFG_TEMPLATE_DIR/rfd.conf /etc/config
-        fi
+	       fi
 
 	if [ ! -e /etc/config/syslog ] ; then
 		cp $CFG_TEMPLATE_DIR/syslog /etc/config
 	fi
 	if [ -e /etc/config/syslog ] ; then
-  		. /etc/config/syslog
+	 		. /etc/config/syslog
 	fi
 }
 
 start() {
 	echo -n "Starting rfd: "
 	init
-	update
+#	update
 	start-stop-daemon -S -q -p /var/run/rfd.pid --exec /bin/rfd -- -d -f /etc/config/rfd.conf -l $LOGLEVEL_RFD
 	echo "OK"
 }
@@ -64,16 +64,16 @@ restart() {
 }
 
 case "$1" in
-  start)
+	 start)
 	start
 	;;
-  stop)
+	 stop)
 	stop
 	;;
-  restart|reload)
+	 restart|reload)
 	restart
 	;;
-  *)
+	 *)
 	echo "Usage: $0 {start|stop|restart}"
 	exit 1
 esac
diff -rupN fs_original/etc/init.d/S99eQ3SystemStarted fs/etc/init.d/S99eQ3SystemStarted
--- fs_original/etc/init.d/S99eQ3SystemStarted	2014-05-21 00:28:40.325846654 +0200
+++ fs/etc/init.d/S99eQ3SystemStarted	2014-05-21 00:33:22.330258409 +0200
@@ -4,32 +4,32 @@
 #
 
 CFG_TEMPLATE_DIR=/etc/config_templates
-SERIAL=$(cat /sys/module/plat_eq3ccu2/parameters/board_serial)
+#SERIAL=$(cat /sys/module/plat_eq3ccu2/parameters/board_serial)
 
 init() {
 	export TZ=`cat /etc/config/TZ`
 
 	# power led on
-	echo 255 > /sys/class/leds/power/brightness
-	echo default-on > /sys/class/leds/power/trigger
+#	echo 255 > /sys/class/leds/power/brightness
+#	echo default-on > /sys/class/leds/power/trigger
 	# internet led off
-	echo none > /sys/class/leds/internet/trigger
+#	echo none > /sys/class/leds/internet/trigger
 	# info led off
-	echo none > /sys/class/leds/info/trigger
-	/bin/hss_led 2>&1 > /dev/null  &
+#	echo none > /sys/class/leds/info/trigger
+#	/bin/hss_led 2>&1 > /dev/null  &
 }
 
 start() {
 	echo -n "System started"
 	init
 	if [ ! -e /etc/config/safemode ] ; then
-        	# meine-homematic.de starten
-        	/opt/mh/startup.sh
+	       	# meine-homematic.de starten
+	       	/opt/mh/startup.sh
 
-        	# Zusatzsoftware starten
-        	run-parts /etc/config/rc.d
+	       	# Zusatzsoftware starten
+	       	run-parts /etc/config/rc.d
 	else
-        	rm -f /etc/config/safemode
+	       	rm -f /etc/config/safemode
 	fi
 	echo "OK"
 }
@@ -43,16 +43,16 @@ stop() {
 }
 
 case "$1" in
-  start)
+	 start)
 	start
 	;;
-  stop)
+	 stop)
 	stop
 	;;
-  restart|reload)
+	 restart|reload)
 	restart
 	;;
-  *)
+	 *)
 	echo "Usage: $0 {start|stop|restart}"
 	exit 1
 esac
diff -rupN fs_original/etc/inittab fs/etc/inittab
--- fs_original/etc/inittab	2014-05-21 00:28:40.475863707 +0200
+++ fs/etc/inittab	2014-05-21 00:33:22.340259199 +0200
@@ -14,27 +14,31 @@
 # process   == program to run
 
 # Startup the system
-null::sysinit:/bin/mount -t proc proc /proc
+#null::sysinit:/bin/mount -t proc proc /proc
 #null::sysinit:/bin/mount -o remount,rw / # REMOUNT_ROOTFS_RW
 null::sysinit:/bin/mkdir -p /dev/pts
 null::sysinit:/bin/mkdir -p /dev/shm
 null::sysinit:/bin/mount -a
 null::sysinit:/bin/hostname -F /etc/hostname
+
+
 # now run any rc scripts
 ::sysinit:/etc/init.d/rcS
 
 # Put a getty on the serial port
-ttyAMA0::respawn:/sbin/getty -L ttyAMA0 115200 vt100 # GENERIC_SERIAL
-ttyGS0::respawn:/sbin/getty -L ttyGS0 115200 vt100 # USB SERIAL
+#ttyAMA0::respawn:/sbin/getty -L ttyAMA0 115200 vt100 # GENERIC_SERIAL
+#ttyGS0::respawn:/sbin/getty -L ttyGS0 115200 vt100 # USB SERIAL
 
 # Stuff to do for the 3-finger salute
 ::ctrlaltdel:/sbin/reboot
 
 # Stuff to do for SIGQUIT 
-::restart:/bin/update_firmware_run
+#::restart:/bin/update_firmware_run
 
 # Stuff to do before rebooting
 ::shutdown:/etc/init.d/rcK
-null::shutdown:/bin/umount -a -r
-null::shutdown:/sbin/swapoff -a
+#null::shutdown:/bin/umount -a -r
+#null::shutdown:/sbin/swapoff -a
 
+# create a console for lxc
+tty1::respawn:/sbin/getty -L tty1 115200 linux
diff -rupN fs_original/opt/mh/startup.sh fs/opt/mh/startup.sh
--- fs_original/opt/mh/startup.sh	2014-05-21 00:28:45.876477466 +0200
+++ fs/opt/mh/startup.sh	2014-05-21 00:33:22.340259199 +0200
@@ -19,9 +19,9 @@ if [ ! -e $USER_DIR ] ; then
 
 fi
 
-mkdir -p /dev/net
-mknod /dev/net/tun c 10 200
-insmod $BASE_DIR/tun.ko
+#mkdir -p /dev/net
+#mknod /dev/net/tun c 10 200
+#insmod $BASE_DIR/tun.ko
 
 
 l1=`ifconfig | grep 'eth0' | tr -s ' ' | cut -d ' ' -f5 | tr ':' '-'`
diff -rupN fs_original/www/api/methods/bidcosrf/setconfiguration-rf.tcl fs/www/api/methods/bidcosrf/setconfiguration-rf.tcl
--- fs_original/www/api/methods/bidcosrf/setconfiguration-rf.tcl	2014-05-21 00:29:22.000575589 +0200
+++ fs/www/api/methods/bidcosrf/setconfiguration-rf.tcl	2014-05-21 00:33:22.350259989 +0200
@@ -20,32 +20,32 @@ set fd [open /etc/config/rfd.conf w]
 puts $fd "# This File was automatically generated"
 puts $fd "# TCP Port for XmlRpc connections"
 puts $fd "Listen Port = 2001"
-puts $fd ""
+#""
 puts $fd "Log Destination = Syslog"
 puts $fd "Log Identifier = rfd"
-puts $fd ""
+#""
 puts $fd "Persist Keys = 1"
-puts $fd ""
+#""
 puts $fd "#PID File = /var/rfd.pid"
 puts $fd "#UDS File = /var/socket_rfd"
-puts $fd ""
+#""
 puts $fd "Device Description Dir = /firmware/rftypes"
 puts $fd "Device Files Dir = /etc/config/rfd"
 puts $fd "Address File = /etc/config/ids"
 puts $fd "Key File = /etc/config/keys"
 puts $fd "Firmware Dir = /firmware"
 puts $fd "Replacemap File = /firmware/rftypes/replaceMap/rfReplaceMap.xml"
-puts $fd ""
-puts $fd ""	
+#""
+#""	
 
 #CCU2 Koprozessor
-puts $fd "\[Interface 0\]"
-puts $fd "Type = CCU2"
-puts $fd "Description = CCU2-Coprocessor"
-puts $fd "ComPortFile = /dev/ttyAPP0"
-puts $fd "AccessFile = /dev/null"
-puts $fd "ResetFile = /dev/ccu2-ic200"
-puts $fd ""
+#puts $fd "\[Interface 0\]"
+#puts $fd "Type = CCU2"
+#puts $fd "Description = CCU2-Coprocessor"
+#puts $fd "ComPortFile = /dev/ttyAPP0"
+#puts $fd "AccessFile = /dev/null"
+#puts $fd "ResetFile = /dev/ccu2-ic200"
+#""
 
 #Lan Interfaces (RF)
 set i 1
diff -rupN fs_original/www/config/control_panel.cgi fs/www/config/control_panel.cgi
--- fs_original/www/config/control_panel.cgi	2014-05-21 00:29:27.361182685 +0200
+++ fs/www/config/control_panel.cgi	2014-05-21 00:33:22.360260779 +0200
@@ -159,10 +159,10 @@ puts {
 
     <!-- Storage Settings -->
     <td>
-      <div class="cpButton">
+      <!-- <div class="cpButton">
         <div  class="StdTableBtn CLASS21701" onclick="showStorageCP();">${btnSysConfStorage}</div>
         <div class="StdTableBtnHelp"><img id="showStorageCPHelp" src="/ise/img/help.png"></div>
-      </div>
+      </div> -->
     </td>
 
 	  </tr>
diff -rupN fs_original/www/config/cp_maintenance.cgi fs/www/config/cp_maintenance.cgi
--- fs_original/www/config/cp_maintenance.cgi	2014-05-21 00:29:27.301175890 +0200
+++ fs/www/config/cp_maintenance.cgi	2014-05-21 00:33:22.370261569 +0200
@@ -318,91 +318,92 @@ proc action_put_page {} {
                 table_data {class="CLASS20904"} {width="400"} {
                     table {class="CLASS20909"} {
                         set cur_version [get_version]
+                        set lxccu_version [read_var "/boot/LXCCUVERSION" VERSION]
                         set serial [get_serial]
                         table_row {
                             table_data {align="left"} {colspan="2" id="actualSWVersion"} {
                                 puts "\${dialogSettingsCMLblActualSoftwareVersion}"
                             }
                             table_data {
-                                puts "$cur_version"
-                            }
-                        }
-                        table_row {
-                            table_data {align="left"} {colspan="2"} {
-                                puts "\${dialogSettingsCMLblAvailableSoftwareVersion}"
-                            }
-                            table_data {id="availableSWVersion"} {
-                                # This doesn´t work properly
-                                # puts [iframe "$REMOTE_FIRMWARE_SCRIPT?cmd=check_version&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2" marginheight=0 marginwidth=0 frameborder=0 width=100 height=20 {scrolling="no"} ]
-                                # The available version will be set further down with "jQuery('#availableSWVersion').html(homematic.com.getLatestVersion());"
-                            }
-                        }
-                        table_row {
-                            table_data {align="left"} {colspan="3"} {
-                                #puts "[bold "Software-Update durchführen"]"
-                                puts "<b>\${dialogSettingsCMLblPerformSoftwareUpdate}</b>"
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep1}"
-                            }
-                            table_data {
-                                division {class="popupControls CLASS20905"} {
-                                    table {
-                                        table_row {
-                                            table_data {
-                                                division {class="CLASS20908"} "onClick=\"window.location.href='$REMOTE_FIRMWARE_SCRIPT?cmd=download&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2';\"" {
-                                                    puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"
-                                                }
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {colspan="2"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep2}"
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {colspan="2"} {
-                                form "$env(SCRIPT_NAME)?sid=$sid" name=firmware_form {target=firmware_upload_iframe} enctype=multipart/form-data method=post {
-                                    export action=firmware_upload
-                                    file_button firmware_file size=30 maxlength=1000000
-                                }
-                                puts {<iframe name="firmware_upload_iframe" style="display: none;"></iframe>}
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep3}"
-                            }
-                            table_data {
-                                division {class="popupControls CLASS20905"} {
-                                    table {
-                                        table_row {
-                                            table_data {
-                                                division {class="CLASS20919"} {onClick="document.firmware_form.submit();showUserHint();"} {
-                                                  puts "\${dialogSettingsCMBtnPerformSoftwareUpdateUpload}"
-                                                }
-                                            }
-                                        }
-                                    }
-                                }
-                            }
-                        }
-                        table_row {
-                            td {width="20"} {}
-                            table_data {align="left"} {colspan="2"} {class="CLASS20920"} {
-                                puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep4}"
+                                puts "$cur_version - LXCCU: $lxccu_version"
                             }
                         }
+                        # table_row {
+                        #     table_data {align="left"} {colspan="2"} {
+                        #         puts "\${dialogSettingsCMLblAvailableSoftwareVersion}"
+                        #     }
+                        #     table_data {id="availableSWVersion"} {
+                        #         # This doesn´t work properly
+                        #         # puts [iframe "$REMOTE_FIRMWARE_SCRIPT?cmd=check_version&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2" marginheight=0 marginwidth=0 frameborder=0 width=100 height=20 {scrolling="no"} ]
+                        #         # The available version will be set further down with "jQuery('#availableSWVersion').html(homematic.com.getLatestVersion());"
+                        #     }
+                        # }
+                        # table_row {
+                        #     table_data {align="left"} {colspan="3"} {
+                        #         #puts "[bold "Software-Update durchführen"]"
+                        #         puts "<b>\${dialogSettingsCMLblPerformSoftwareUpdate}</b>"
+                        #     }
+                        # }
+                        # table_row {
+                        #     td {width="20"} {}
+                        #     table_data {align="left"} {
+                        #         puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep1}"
+                        #     }
+                        #     table_data {
+                        #         division {class="popupControls CLASS20905"} {
+                        #             table {
+                        #                 table_row {
+                        #                     table_data {
+                        #                         division {class="CLASS20908"} "onClick=\"window.location.href='$REMOTE_FIRMWARE_SCRIPT?cmd=download&version=$cur_version&serial=$serial&lang=de&product=HM-CCU2';\"" {
+                        #                             puts "\${dialogSettingsCMBtnPerformSoftwareUpdateDownload}"
+                        #                         }
+                        #                     }
+                        #                 }
+                        #             }
+                        #         }
+                        #     }
+                        # }
+                        # table_row {
+                        #     td {width="20"} {}
+                        #     table_data {align="left"} {colspan="2"} {
+                        #         puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep2}"
+                        #     }
+                        # }
+                        # table_row {
+                        #     td {width="20"} {}
+                        #     table_data {colspan="2"} {
+                        #         form "$env(SCRIPT_NAME)?sid=$sid" name=firmware_form {target=firmware_upload_iframe} enctype=multipart/form-data method=post {
+                        #             export action=firmware_upload
+                        #             file_button firmware_file size=30 maxlength=1000000
+                        #         }
+                        #         puts {<iframe name="firmware_upload_iframe" style="display: none;"></iframe>}
+                        #     }
+                        # }
+                        # table_row {
+                        #     td {width="20"} {}
+                        #     table_data {align="left"} {
+                        #         puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep3}"
+                        #     }
+                        #     table_data {
+                        #         division {class="popupControls CLASS20905"} {
+                        #             table {
+                        #                 table_row {
+                        #                     table_data {
+                        #                         division {class="CLASS20919"} {onClick="document.firmware_form.submit();showUserHint();"} {
+                        #                           puts "\${dialogSettingsCMBtnPerformSoftwareUpdateUpload}"
+                        #                         }
+                        #                     }
+                        #                 }
+                        #             }
+                        #         }
+                        #     }
+                        # }
+                        # table_row {
+                        #     td {width="20"} {}
+                        #     table_data {align="left"} {colspan="2"} {class="CLASS20920"} {
+                        #         puts "\${dialogSettingsCMLblPerformSoftwareUpdateStep4}"
+                        #     }
+                        # }
                     }
                 }
                 table_data {align="left"} {class="CLASS20921"} {
diff -rupN fs_original/www/config/cp_security.cgi fs/www/config/cp_security.cgi
--- fs_original/www/config/cp_security.cgi	2014-05-21 00:29:27.351181552 +0200
+++ fs/www/config/cp_security.cgi	2014-05-21 00:34:43.706845215 +0200
@@ -41,30 +41,30 @@ proc get_current_key_index {} {
   if { [info exists keys($CURRENT_INDEX)] } then {
     set currentIndex $keys($CURRENT_INDEX)
   }
-  
+
   return $currentIndex
 }
 
 proc set_key { key } {
   global RFD_URL
-  
+
   xmlrpc $RFD_URL changeKey $key
-  
+
   # set always new key with crypttool
   set index [get_current_key_index]
   incr index 1
-    
+
   set fd [open "|crypttool -S -k $key -i $index" r]
   close $fd
 }
 
 proc action_change_key {} {
     global env RFD_URL
-    
+
     http_head
     import key1
-    import key2    
-    
+    import key2
+
     if { "$key1" != "$key2" } {
         #put_message "\${dialogSettingsSecurityMessageErrorSecKeyTitle}" "Die beiden eingegebenen Schl&uuml;ssel stimmen nicht &uuml;berein." {\${dialogBack} "showSecurityCP();"}
         put_message "\${dialogSettingsSecurityMessageErrorSecKeyTitle}" "\${dialogSettingsSecurityMessageErrorSecKeyContentKeysNotIdentical}" {\${dialogBack} "showSecurityCP();"}
@@ -75,13 +75,13 @@ proc action_change_key {} {
         put_message "\${dialogSettingsSecurityMessageErrorSecKeyTitle}" \${dialogSettingsSecurityMessageErrorSecKeyContentKeyShort} {\${dialogBack} "showSecurityCP();"}
         return
     }
-		
+
 		if { 0 == [regexp {^[0-9a-zA-Z_]+$} $key1 dummy] } {
         #put_message "Sicherheitsschl&uuml;ssel setzen - Fehler" "Der eingegebene Schl&uuml;ssel darf keine Sonderzeichen enthalten. Erlaubt sind lediglich die Buchstaben A bis Z, die Ziffern 0 bis 9 sowie der Unterstrich." {"Zur&uuml;ck" "showSecurityCP();"}
         put_message "\${dialogSettingsSecurityMessageErrorSecKeyTitle}" \${dialogSettingsSecurityMessageErrorSecKeyContentIllegalChar} {\${dialogBack} "showSecurityCP();"}
         return
 		}
-		
+
     # check the entered key against our current system key
     if { ![catch {exec crypttool -v -t 3 -k "$key1"}]} {
 
@@ -90,7 +90,7 @@ proc action_change_key {} {
         put_message "\${dialogSettingsSecurityMessageHintSecKeyTitle}" \${dialogSettingsSecurityMessageErrorSecKeyContentKeysIsIdentical} {\${dialogBack} "showSecurityCP();"}
         return
     }
-    
+
     if { [catch {set_key $key1}] } {
         # "Der Schlüssel konnte nicht gesetzt werden. Das liegt vermutlich daran, daß "
         # "der aktuelle Schlü;ssel noch nicht an alle Komponenten übertragen wurde. "
@@ -103,7 +103,7 @@ proc action_change_key {} {
 
 proc action_factory_reset_check {} {
    global env
-   
+
     http_head
     division {class="popupTitle"} {
         puts "\${dialogSettingsSecurityMessagePerformSystemResetTitle}"
@@ -136,7 +136,7 @@ proc action_factory_reset_check {} {
                                     cgi_put "<input type=hidden name=\"key\" value=dummy id=\"text_key\"/>"
                                 }
                             }
-                        }                        
+                        }
                         table_row {
                             table_data {align="right"} {class="CLASS20812"} {colspan="3"} {
                                 division {class="popupControls CLASS20811"} {
@@ -196,12 +196,13 @@ proc action_factory_reset_go {} {
     }
     if { [catch {
         exec crypttool -r
-        # exec umount /usr/local
-        # exec /usr/sbin/ubidetach -p /dev/mtd6
-        # exec /usr/sbin/ubiformat /dev/mtd6 -y
-        # exec /usr/sbin/ubiattach -p /dev/mtd6
-        # exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
-        # exec mount /usr/local
+        # #exec umount /usr/local
+        # #exec /usr/sbin/ubidetach -p /dev/mtd6
+        # #exec /usr/sbin/ubiformat /dev/mtd6 -y
+        # #exec /usr/sbin/ubiattach -p /dev/mtd6
+        # #exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
+        # #exec mount /usr/local
+exec rm -rf /usr/local/*
         exec touch /var/doFactoryReset
         exec kill -SIGQUIT 1
     }]} {
@@ -279,27 +280,27 @@ proc action_factory_reset_go {} {
 proc action_backup_restore_check {} {
     global env
     cd /tmp/
-    
+
     http_head
     set i 0
     if { [catch {
         exec tar xf new_config.tar
         file delete -force /tmp/new_config.tar
-		
+
     set config_version [read_version "firmware_version"]
 		set ccu1_backup false
 		if { [version_compare $config_version 2.0.0] < 0 } {
-			set ccu1_backup true		
+			set ccu1_backup true
 		}
-		
+
         set system_has_user_key [catch {exec crypttool -v -t 0}]
         set stored_signature [exec cat signature]
         set calculated_signature [exec crypttool -s -t 0 <usr_local.tar.gz]
-		if { "false" == $ccu1_backup } {				
+		if { "false" == $ccu1_backup } {
 			set config_has_user_key [expr {"$stored_signature" != "$calculated_signature"} ]
 		} else {	#CCU1 used an other default key. So we can´t check with crypttool.
 			set config_has_user_key [exec cat key_index]
-		}		
+		}
     } errorMessage] } {
         # "Die ausgewählte Datei enthält kein HomeMatic Systembackup. "
         # "Bitte wählen Sie eine gültige Backup-Datei aus."
@@ -361,7 +362,7 @@ proc action_backup_restore_check {} {
                                     cgi_put "<input type=hidden name=\"key\" value=dummy id=\"text_key\"/>"
                                 }
                             }
-                        }                        
+                        }
                         table_row {
                             table_data {align="right"} {class="CLASS20812"} {colspan="3"} {
                                 division {class="popupControls CLASS20811"} {
@@ -415,7 +416,7 @@ proc action_backup_restore_go {} {
     global env
     cd /tmp/
     http_head
-    
+
     set system_version [read_version "/boot/VERSION"]
     set config_version [read_version "firmware_version"]
 	set ccu1_backup false
@@ -427,10 +428,10 @@ proc action_backup_restore_go {} {
             } {"OK" }
     }
   }
-	
+
     set system_has_user_key [catch {exec crypttool -v -t 0}]
     set stored_signature [exec cat signature]
-	if { "false" == $ccu1_backup } {	
+	if { "false" == $ccu1_backup } {
 		set config_has_user_key [expr {"$stored_signature" != "[exec crypttool -s -t 0 <usr_local.tar.gz]"} ]
 	} else {	#CCU1 used an other default key. So we can´t check with crypttool.
 		set config_has_user_key [exec cat key_index]
@@ -451,7 +452,7 @@ proc action_backup_restore_go {} {
             return
         }
     }
-    
+
     if { [version_compare $config_version $system_version] > 0 } {
         # set msg "Das Einspielen des Backups ist nicht m&ouml;glich. Das vorliegende Backup basiert auf der Zentralen-Firmware $config_version.<br>\n"
         # append msg "Diese Firmware ist aktueller, als die derzeit auf der Zentrale installierte Version ($system_version).<br>\n"
@@ -460,26 +461,27 @@ proc action_backup_restore_go {} {
         put_message "\${dialogSettingsSecurityMessageSysBackupFWUpdateNecessaryTitle}" "\${dialogSettingsSecurityMessageSysBackupFWUpdateNecessaryContentA} $config_version \${dialogSettingsSecurityMessageSysBackupFWUpdateNecessaryContentB} ($system_version) \${dialogSettingsSecurityMessageSysBackupFWUpdateNecessaryContentC}"
         return
     }
-    
+
     #get key index
     set stored_index [exec cat key_index]
-    if { !$system_has_user_key && $config_has_user_key } {        
+    if { !$system_has_user_key && $config_has_user_key } {
         exec crypttool -S -i $stored_index -k "$key"
     }
-    
+
     cd /
     catch {
         exec run-parts -a stop /etc/config/rc.d
 	exec killall crond
     }
-		
+
 	if { "false" == $ccu1_backup } {	# backup for version >= 2
-		exec umount /usr/local
-        	exec /usr/sbin/ubidetach -p /dev/mtd6
-        	exec /usr/sbin/ubiformat /dev/mtd6 -y
-        	exec /usr/sbin/ubiattach -p /dev/mtd6
-        	exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
-        	exec mount /usr/local
+		#exec umount /usr/local
+        	#exec /usr/sbin/ubidetach -p /dev/mtd6
+        	#exec /usr/sbin/ubiformat /dev/mtd6 -y
+        	#exec /usr/sbin/ubiattach -p /dev/mtd6
+        	#exec /usr/sbin/ubimkvol /dev/ubi1 -N user -m
+        	#exec mount /usr/local
+exec rm -rf /usr/local/*
 
 		if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
 			# set msg "Beim Einspielen des Systembackups ist ein Fehler aufgetreten. Bitte versuchen Sie es erneut. "
@@ -489,16 +491,16 @@ proc action_backup_restore_go {} {
 			set backuperror true
 		} else {
 			set backuperror false
-		}		
-	} else {	# backup for version < 2			
+		}
+	} else {	# backup for version < 2
 		#delete existing files
 		file delete -force /tmp/backup
-		file delete -force /etc/config/hs485d /etc/config/hs485types /etc/config/rfd /etc/config/userprofiles 
+		file delete -force /etc/config/hs485d /etc/config/hs485types /etc/config/rfd /etc/config/userprofiles
 		file delete -force /etc/config/homematic.regadom /etc/config/homematic.regadom.bak
-		file delete -force /etc/config/ids /etc/config/ntpclient /etc/config/rega.conf /etc/config/syslog 
+		file delete -force /etc/config/ids /etc/config/ntpclient /etc/config/rega.conf /etc/config/syslog
 		file delete -force /etc/config/tweaks /etc/config/TZ /etc/config/server.pem /etc/config/keys
 		file delete -force /etc/config/time.conf
-		
+
 		file mkdir /tmp/backup
 		cd /tmp/backup
 		if { [catch {exec tar xzf /tmp/usr_local.tar.gz} errorMessage] } {
@@ -509,8 +511,8 @@ proc action_backup_restore_go {} {
 			if { [file exists /tmp/backup/usr/local/etc/config] } {
 				cd /tmp/backup/usr/local/etc/config
 			}
-			
-			if { [file exists hs485d] } { file copy hs485d /etc/config/hs485d }				
+
+			if { [file exists hs485d] } { file copy hs485d /etc/config/hs485d }
 			if { [file exists hs485types] } { file copy hs485types /etc/config/hs485types }
 			if { [file exists rfd] } { file copy rfd /etc/config/rfd }
 			if { [file exists userprofiles] } { file copy userprofiles /etc/config/userprofiles }
@@ -522,12 +524,12 @@ proc action_backup_restore_go {} {
 			if { [file exists syslog] } { file copy syslog /etc/config/syslog }
 			if { [file exists tweaks] } { file copy tweaks /etc/config/tweaks }
 			if { [file exists TZ] } { file copy TZ /etc/config/TZ }
-			if { [file exists server.pem] } { file copy server.pem /etc/config/server.pem }			
+			if { [file exists server.pem] } { file copy server.pem /etc/config/server.pem }
 			if { [file exists time.conf] } { file copy time.conf /etc/config/time.conf }
 			if { [file exists keys] } then {
 				file copy keys /etc/config/keys
 			} else {
-				if { $config_has_user_key } then {          
+				if { $config_has_user_key } then {
 					set fd [open "/etc/config/crypttool.cfg" r]
 					set content [read $fd]
 					close $fd
@@ -537,16 +539,16 @@ proc action_backup_restore_go {} {
 						if { [regexp {([0-9]+) (.*)} $line dummy key value] } then {
 							set key [string trim $key]
 							set value [string trim $value]
-							set keys($key) $value						
+							set keys($key) $value
 						}
 					}
-          					
+
 					if { [info exists keys($stored_index)] } then {
-						set keyvalue $keys($stored_index)							
+						set keyvalue $keys($stored_index)
 						set fd -1
-            
+
 						catch {set fd [open "/etc/config/keys" w]}
-            
+
 						if { $fd > 0 } {
 								puts $fd "Current Index = $stored_index"
 								puts $fd "Key 0 ="
@@ -554,20 +556,20 @@ proc action_backup_restore_go {} {
 								puts $fd "Last Index = 0"
 								close $fd
 						}
-					}          
-				}				
+					}
+				}
 			}
 
 			catch {exec eq3configcmd rfd-interface-copy rfd.conf /etc/config/rfd.conf}
 
 			set backuperror false
-		}					
+		}
 		cd /
 	}
-	
+
 	if { "false" == $backuperror } {
-        exec mount -o remount,ro /usr/local
-        exec mount -o remount,rw /usr/local
+        #exec mount -o remount,ro /usr/local
+        #exec mount -o remount,rw /usr/local
         division {class="popupTitle"} {
             puts "\${dialogSettingsSecurityMessageSysBackupRestartSystemTitle}"
         }
@@ -646,16 +648,16 @@ proc action_set_session_timeout {} {
   set REGA_CONF_FILE "/etc/config/rega.conf"
 
   http_head
-  
+
   if { [catch {
     import timeout
     if { $timeout != [expr int($timeout)] || 180 > $timeout || 600 < $timeout } then { error "invalid timeout" }
-    
+
     set    rega_conf "# rega.conf\n"
     append rega_conf "# This file is generated by cp_security.cgi\n"
     append rega_conf "SessionTimeout=$timeout\n"
     saveToFile $REGA_CONF_FILE rega_conf
-    
+
 
     # Die neue Zeit bis zum Ablauf der Sitzung wurde erfolgreich &uuml;bernommen.
     # Die &Auml;nderung wird mit dem n&auml;chsten Start der HomeMatic Zentrale wirksam.
@@ -664,7 +666,7 @@ proc action_set_session_timeout {} {
         ${dialogSettingsSecurityMessageSessionTimeOutSaveContent}
       </p>
     }
-    
+
   }]} then {
     # Fehler: Session Timeout &uuml;bernehmen
     # Bitte w&auml;hlen Sie eine Zeit zwischen 180 und 600 Sekunden.
@@ -808,38 +810,40 @@ proc action_put_page {} {
                     }
                 }
             }
-            table_row {class="CLASS20806"} {
-                table_data {class="CLASS20807"} {
-                    puts "\${dialogSettingsSecurityTDSysReset}"
-                }
-                table_data {class="CLASS20808"} {
-                    table {class="CLASS20810"} {width="100%"} {
-                        table_row {
-                            table_data {align="left"} {colspan="2"} {
-                                puts "\${dialogSettingsSecurityLblSysResetPerform}"
-                            }
-                            table_data {align="right"} {
-                                division {class="popupControls CLASS20811"} {
-                                    division {class="CLASS20818 colorGradient50px"} {onClick="OnFactoryReset()"} {
-                                        puts "\${dialogSettingsSecurityBtnSysReset}"
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-                table_data {align="left"} {class="CLASS20816"} {
-                    puts "\${dialogSettingsSecurityHintSysReset1}"
-                    br
-                    #puts "Die Zentrale wird in den Werkszustand zur&uuml;ckgesetzt. Alle angelernten Ger&auml;te und"
-                    #puts "alle erstellten Programme werden gel&ouml;scht"
-                    puts "\${dialogSettingsSecurityHintSysReset2}"
-                    br
-                    #puts "Alle Ger&auml;tekonfigurationen und alle mit &quot;Direkte Verkn&uuml;pfungen&quot; angelegten"
-                    #puts "Ger&auml;teverkn&uuml;pfungen bleiben bestehen und sind funktionsf&auml;hig."
-                    puts "\${dialogSettingsSecurityHintSysReset3}"
-                }
-            }
+# begin: don't need because know you use dpkg-reconfigure lxccu
+#            table_row {class="CLASS20806"} {
+#                table_data {class="CLASS20807"} {
+#                    puts "\${dialogSettingsSecurityTDSysReset}"
+#                }
+#                table_data {class="CLASS20808"} {
+#                    table {class="CLASS20810"} {width="100%"} {
+#                        table_row {
+#                            table_data {align="left"} {colspan="2"} {
+#                                puts "\${dialogSettingsSecurityLblSysResetPerform}"
+#                            }
+#                            table_data {align="right"} {
+#                                division {class="popupControls CLASS20811"} {
+#                                    division {class="CLASS20818 colorGradient50px"} {onClick="OnFactoryReset()"} {
+#                                        puts "\${dialogSettingsSecurityBtnSysReset}"
+#                                    }
+#                                }
+#                            }
+#                        }
+#                    }
+#                }
+#                table_data {align="left"} {class="CLASS20816"} {
+#                    puts "\${dialogSettingsSecurityHintSysReset1}"
+#                    br
+#                    #puts "Die Zentrale wird in den Werkszustand zur&uuml;ckgesetzt. Alle angelernten Ger&auml;te und"
+#                    #puts "alle erstellten Programme werden gel&ouml;scht"
+#                    puts "\${dialogSettingsSecurityHintSysReset2}"
+#                    br
+#                    #puts "Alle Ger&auml;tekonfigurationen und alle mit &quot;Direkte Verkn&uuml;pfungen&quot; angelegten"
+#                    #puts "Ger&auml;teverkn&uuml;pfungen bleiben bestehen und sind funktionsf&auml;hig."
+#                    puts "\${dialogSettingsSecurityHintSysReset3}"
+#                }
+#            }
+# end
           table_row {class="CLASS20806"} {
             table_data {class="CLASS20807"} {
               puts "\${dialogSettingsSecurityTDSessionTimeout}"
@@ -965,7 +969,7 @@ proc action_put_page {} {
                 //ProgressBar = new ProgressBarMsgBox("Systembackup wird übertragen...", 1);
                 //ProgressBar.show();
                 //ProgressBar.StartKnightRiderLight();
-            
+
                 document.backup_form.submit();
             };
             OnSetSessionTimeout = function() {
@@ -1078,7 +1082,7 @@ proc action_create_backup {} {
 proc action_backup_upload {} {
     global env sid
     cd /tmp/
-    
+
     http_head
     import_file -client backup_file
     file rename -force -- [lindex $backup_file 0] "/tmp/new_config.tar"
diff -rupN fs_original/www/config/cp_software.cgi fs/www/config/cp_software.cgi
--- fs_original/www/config/cp_software.cgi	2014-05-21 00:29:27.231167964 +0200
+++ fs/www/config/cp_software.cgi	2014-05-21 00:33:22.390263148 +0200
@@ -507,7 +507,8 @@ proc action_install_start {} {
     if {[isOldCCU]} {
         exec /sbin/init -q
     } else {
-        exec /bin/kill -SIGQUIT 1
+        #exec /bin/kill -SIGQUIT 1
+        exec /sbin/reboot
     }
 }
 
